plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/ProductCollectionController.php

```diff
@@ -1939,12 +819,14 @@ protected function register_collection_handlers( $collection_name, $build_query,
 			throw new \InvalidArgumentException( 'Collection handlers already registered for ' . esc_html( $collection_name ) );
 		}
 
-		$this->collection_handler_store[ $collection_name ] = array(
+		$handlers = array(
 			'build_query'   => $build_query,
 			'frontend_args' => $frontend_args,
 			'editor_args'   => $editor_args,
 			'preview_query' => $preview_query,
 		);
+		$this->collection_handler_store[ $collection_name ] = $handlers;
+		$this->query_builder->set_collection_handler_store( $collection_name, $handlers );
```

Comment:
We want to set the `collection_handler_store` in the `ProductQueryBuilder` class too.